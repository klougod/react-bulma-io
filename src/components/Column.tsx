import React from 'react'
import PropTypes from 'prop-types'

export const Column = ({
  size,
  tablet,
  mobile,
  desktop,
  children,
  className
}: any) => {
  let renderSize = size ? `is-${size}` : ''
  renderSize = tablet ? `${renderSize} is-${tablet}-tablet` : renderSize
  renderSize = mobile ? `${renderSize} is-${mobile}-mobile` : renderSize
  renderSize = desktop ? `${renderSize} is-${desktop}-desktop` : renderSize

  return <div className={`column ${renderSize} ${className}`}>{children}</div>
}

Column.defaultProps = {
  size: undefined,
  tablet: undefined,
  mobile: undefined,
  desktop: undefined,
  className: ''
}

Column.propTypes = {
  tablet: PropTypes.number,
  mobile: PropTypes.number,
  desktop: PropTypes.number,
  size: PropTypes.number,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default Column
