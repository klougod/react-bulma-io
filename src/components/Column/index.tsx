import React from 'react'
import PropTypes from 'prop-types'

export const Column = ({
  size,
  tablet,
  mobile,
  desktop,
  children,
  className
}: any) => {
  let renderSize = size ? `is-${size}` : ''
  renderSize = tablet ? `${renderSize} is-${tablet}-tablet` : renderSize
  renderSize = mobile ? `${renderSize} is-${mobile}-mobile` : renderSize
  renderSize = desktop ? `${renderSize} is-${desktop}-desktop` : renderSize

  return <div className={`column ${renderSize} ${className}`}>{children}</div>
}

Column.defaultProps = {
  size: null,
  tablet: null,
  mobile: null,
  desktop: null,
  className: ''
}

Column.propTypes = {
  tablet: PropTypes.number,
  mobile: PropTypes.number,
  desktop: PropTypes.number,
  size: PropTypes.number,
  className: PropTypes.string,
  children: PropTypes.node
}

export default Column
