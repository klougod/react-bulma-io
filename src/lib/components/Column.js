import React from 'react'
import PropTypes from 'prop-types'

export const Column = props => {

  const {
    size,
    children,
    className
  } = props

  return (
    <div className={`column is-${size} ${className}`}>
      {children}
    </div>
  )
}

Column.defaultProps = {
  size: 12,
  className: ''
}

Column.propTypes = {
  size: PropTypes.number,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default Column