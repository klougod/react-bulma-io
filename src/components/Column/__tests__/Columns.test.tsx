import React from 'react'
import Enzyme from 'enzyme'

import Column from '../'

describe('Column component test', function () {
  it('should render without crash without props', function () {
    const columnMock = Enzyme.shallow(<Column />)
    expect(columnMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const columnMockProps = {
      className: 'columnCLass',
      size: 2,
      mobile: 3,
      tablet: 4,
      desktop: 5,
    }
    const columnMock = Enzyme.shallow(<Column {...columnMockProps} />)
    expect(columnMock).toMatchSnapshot()
  })
})
