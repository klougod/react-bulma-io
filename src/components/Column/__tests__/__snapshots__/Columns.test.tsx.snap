// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column component test should render without crash with props 1`] = `
<div
  className="column is-2 is-4-tablet is-3-mobile is-5-desktop columnCLass"
/>
`;

exports[`Column component test should render without crash without props 1`] = `
<div
  className="column  "
/>
`;
